tests/console/docker_image.pm

```diff
@@ -55,54 +59,59 @@ sub run {
     }
 
     for my $i (0 .. $#$image_names) {
-        # Load the image
-        assert_script_run("docker pull $image_names->[$i]", 1000);
-        # Running executables works
-        assert_script_run qq{docker container run --entrypoint '/bin/bash' --rm $image_names->[$i] -c 'echo "I work" | grep "I work"'};
-        # It is the right version
-        if (is_sle) {
-            $osversion = get_required_var("VERSION") =~ s/-SP/ SP/r;    # 15 -> 15, 15-SP1 -> 15 SP1
-            validate_script_output("docker container run --entrypoint '/bin/bash' --rm $image_names->[$i] -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $osversion"/ });
+        # Ignore internal images on public cloud
+        if (isPublicCloud && $image_names->[$i] =~ /registry.suse.[de|com]/) {
```

Comment:
we looping here over list which we got from `get_suse_container_urls` this function already contains quite some logic to filter out what images we taking and which we skipping . I think better move this condition there and drop this `next` here

Reply:
Modified `get_suse_container_urls` accordingly. Good idea!

Reply:
@asmorodskyi we had to revert the changes in `get_suse_container_urls` as it was causing some problems. We keep this for a future PR after refactoring `docker_image` and `podman_image`